sp/src/game/client/mapbase/vscript_vgui.cpp

```diff
@@ -0,0 +1,1871 @@
+		if ( idx != g_ScriptFonts.InvalidIndex() )
+		{
+			font = g_ScriptFonts[idx].font;
+		}
+	}
+	else
+	{
+		HScheme sch = scheme()->GetScheme( schema );
+		font = scheme()->GetIScheme(sch)->GetFont( name, proportional );
+
+		// Update known count
+		if ( font > (unsigned int)g_nFontCount )
+			g_nFontCount = font;
+	}
+
+	return HandleToInt( font );
+}
+
+void CScriptSurface::SetFont( int font )
+{
+	surface()->DrawSetTextFont( IntToFontHandle(font) );
+}
+
+int CScriptSurface::GetTextWidth( int font, const char* text )
+{
+	int w, t;
+	wchar_t wcs[512];
+	g_pVGuiLocalize->ConvertANSIToUnicode( text, wcs, sizeof(wcs) );
+	surface()->GetTextSize( IntToFontHandle(font), wcs, w, t );
+	return w;
+}
+
+int CScriptSurface::GetFontTall( int font )
+{
+	return surface()->GetFontTall( IntToFontHandle(font) );
+}
+
+int CScriptSurface::GetCharacterWidth( int font, int ch )
+{
+	return surface()->GetCharacterWidth( IntToFontHandle(font), ch );
+}
+
+int CScriptSurface::CreateFont( const char *customName, const char *windowsFontName, int tall, int weight, int blur, int scanlines, int flags, bool proportional )
+{
+	HFont font;
+	const char *fontAlias = GetFixedFontName( customName, proportional );
+
+	int idx = g_ScriptFonts.Find( fontAlias );
+	if ( idx != g_ScriptFonts.InvalidIndex() )
+	{
+		font = g_ScriptFonts[idx].font;
+
+		int oldTall = surface()->GetFontTall( font );
+		if ( proportional )
+			oldTall = scheme()->GetProportionalNormalizedValue( oldTall );
+
+		if ( oldTall == tall &&
+			!V_stricmp( surface()->GetFontName( font ), windowsFontName ) )
+		{
+			// Name and height match, this is most likely the same font.
+			return font;
+		}
+
+#ifdef _DEBUG
+		Msg( "CreateFont called on existing font [%i]%s\n", font, fontAlias );
+#endif
+	}
+	else
+	{
+		int size = V_strlen( windowsFontName ) + 1;
+		char *fontName = new char[ size ];
+		V_strncpy( fontName, windowsFontName, size );
+
+		font = surface()->CreateFont();
+
+		// Sanity check
+		Assert( font > (unsigned int)g_nFontCount && font < INT_MAX );
+
+		// Update known count
+		if ( font > (unsigned int)g_nFontCount )
+			g_nFontCount = font;
+
+		FontData_t &data = g_ScriptFonts.Element( g_ScriptFonts.Insert( fontAlias ) );
+		data.font = font;
+		data.name = fontName;
+		data.tall = tall;
+		data.weight = weight;
+		data.blur = blur;
+		data.scanlines = scanlines;
+		data.flags = flags;
+		data.proportional = proportional;
+	}
+
+	if ( proportional )
+	{
+		tall = scheme()->GetProportionalScaledValue( tall );
+		blur = scheme()->GetProportionalScaledValue( blur );
+		scanlines = scheme()->GetProportionalScaledValue( scanlines );
+	}
+
+#ifdef _DEBUG
+	bool bSuccess =
+#endif
+
+		surface()->SetFontGlyphSet( font, windowsFontName, tall, weight, blur, scanlines, flags );
+
+#ifdef _DEBUG
+	if ( bSuccess )
+	{
+		Msg( "Create font [%i]%s\n", font, fontAlias );
+	}
+	else
+	{
+		Warning( "Failed to create font [%i]%s\n", font, fontAlias );
+	}
+#endif
```

Comment:
Indentation looks weird. May I suggest
```suggestion
	bool bSuccess = surface()->SetFontGlyphSet( font, windowsFontName, tall, weight, blur, scanlines, flags );

#ifdef _DEBUG
	if ( bSuccess )
	{
		Msg( "Create font [%i]%s\n", font, fontAlias );
	}
	else
	{
		Warning( "Failed to create font [%i]%s\n", font, fontAlias );
	}
#else
	(void)bSuccess;
#endif
```